File: examples/javascript/delete-subject.mjs
import { memside } from "./memside-client.mjs";

const [subjectId, confirmation] = process.argv.slice(2);
if (!subjectId || !confirmation) {
  throw new Error(
    "Usage: node delete-subject.mjs <subject-id> <exact-confirmation>"
  );
}

const result = await memside(`/subjects/${encodeURIComponent(subjectId)}`, {
  method: "DELETE",
  body: JSON.stringify({ delete_confirmation: confirmation }),
});

console.log(JSON.stringify(result, null, 2));

File: examples/python/delete_subject.py
import json
import sys
from urllib.parse import quote

from memside_http import api_request


if len(sys.argv) != 3:
    raise RuntimeError(
        "Usage: python delete_subject.py "
        "<subject-id> <exact-confirmation>"
    )

subject_id, confirmation = sys.argv[1:]
result = api_request(
    "DELETE",
    f"/subjects/{quote(subject_id, safe='')}",
    body={"delete_confirmation": confirmation},
)

print(json.dumps(result, indent=2))
